[SPARK-57699][DOCS] Update docs/control-flow/leave-stmt.md (#56329)
apache/spark: docs/control-flow/leave-stmt.md

  limitations under the License.
---

> Terminates the execution of an iteration of a looping statement and exits the loop.

Reviewer: The new description still reads "Terminates the execution of an iteration of a looping statement and exits the loop." This is inherited ITERATE phrasing and is semantically incomplete for LEAVE: verified against AstBuilder.scala leaveOrIterateContextHasLabel (lines 607-635), LEAVE (isIterate=false) also matches BeginEndCompoundBlockContext and exits labeled BEGIN...END compound blocks, not only loops (ITERATE throws for that case).

Suggesting something like:
"Exits the execution of a labeled statement, which may be a looping construct or a labeled compound block."

This should be strictly better than the wrong original, but still inaccurate as shipped.

> Exits the execution of a labeled statement, which may be a looping construct or a labeled compound block.

Reviewer 2: I'd rephrase this a bit with a more strict wording - we either "exit the statement" or "terminate the execution".
Also, I think that the we are using "statement" throughout the docs for all the constructs - so, "looping statement" and "compound statement" seem more consistent.
Thus, I'd do something like:

```suggestion
Exits a labeled looping statement or compound statement, transferring control to the statement that follows it.
```
or a simpler version
```suggestion
Terminates the execution of a labeled looping or compound statement.
```
or whatever else you can think of that fits the best.

> - [WHILE Statement](../control-flow/while-stmt.html)

Reviewer 2: nit: while we are at it, let's also add a link to REPEAT statement here - other related statements have it.
